pinot-controller/src/main/java/org/apache/pinot/controller/util/TableSizeReader.java

```diff
@@ -259,6 +264,10 @@ public TableSubTypeSizeDetails getTableSubtypeSize(String tableNameWithType, int
       String server = entry.getKey();
       List<SegmentSizeInfo> segmentSizeInfoList = serverToSegmentSizeInfoListMap.get(server);
       if (segmentSizeInfoList != null) {
+        segmentSizeInfoList = serverToSegmentSizeInfoListMap.get(server)
+            .stream()
+            .filter(segmentSizeInfo -> entry.getValue().contains(segmentSizeInfo.getSegmentName()))
+            .collect(Collectors.toList());
```

Comment:
Is this necessary?

Reply:
yes, a final `collect.toList` is needed

Reply:
Currently this filtering logic is very inefficient because you are doing `contains` with a `List`. Given we need to filter anyway, we can optimize it by:
1. Create a map from segment name to segment size from `segmentSizeInfoList`
2. Loop over the `segments` (`entry.getValue()`), check if segment is available from the map

You may also remove the TODO on line 262 after making this change